core/jobs/batch_jobs/firebase_sync_jobs.py

```
@@ -0,0 +1,142 @@
+class FirebaseSyncRecordsJob(base_jobs.JobBase):
+    """Sync Firebase records to match with Oppia's user & auth models."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        if (
+            feature_flag_domain.get_server_mode()
+            == feature_flag_domain.ServerMode.PROD
+        ):
+            job_name = self.__class__.__name__
+            raise PermissionError(f'{job_name} must never be run in production')
+
+        key_by_email = lambda record: (record.email, record)
```

Ruff (0.15.15)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `key_by_email` as a `def`

(E731)